api/v1beta2/ibmvpcmachine_types.go

```diff
@@ -36,10 +36,8 @@ type IBMVPCMachineSpec struct {
 	// Name of the instance.
 	Name string `json:"name,omitempty"`
 
-	// Image is the id of OS image which would be install on the instance.
-	// Example: r134-ed3f775f-ad7e-4e37-ae62-7199b4988b00
-	// TODO: allow user to specify a image name is much reasonable. Example: ibm-ubuntu-18-04-1-minimal-amd64-2
-	Image string `json:"image"`
+	// Image is the OS image which would be install on the instance.
+	Image *IBMVPCResourceReference `json:"image"`
```

Comment:
what will happen when user supply both ID and Name? can we add that information?

Reply:
lets add some test cases as well for this to test it.

Reply:
Added note, also added test case for the use-case.